Regular-Question-Answers/jump-game.cpp

```diff
@@ -0,0 +1,48 @@
+// Author: Ali Emre Kaya
+// Reviewer:
+// Question Link: https://leetcode.com/problems/jump-game/
+
+// Time Complexity: O(n)
+// Space Complexity: O(1)
+
+class Solution {
+public:
+    bool canJump(vector<int>& nums) {
+        // if size of the array 1, directly return true
+        if(nums.size() == 1) return true;
+
+        // current index you're standing at
+        int i = 0;
+        // each array element can be visit at most one time
+        //so time complexity is O(n)
+        while (i < nums.size()) {
```

Comment:
The code is readable but you can change the variables, "i" means "current index" therefore curr_idx, and "j" means "reachable index", therefore "reach_idx".